### PR TITLE
fix(select): fix allowCreate and tokenSeparators property is conflict

### PR DESCRIPTION
**Pull request by chenc041**
## Types of changes

- [ ] New feature
- [x] Bug fix
- [ ] Documentation change
- [ ] Coding style change
- [ ] Component style change
- [ ] Refactoring
- [ ] Performance improvement
- [ ] Test cases
- [ ] Continuous integration
- [ ] Typescript definition change
- [ ] Breaking change
- [ ] Others 

## Background and context
The ```allowCreate``` property will not working,  when set ```allowCreate``` and ```tokenSeparators```  together

## Solution
Always  ```allowCreate``` first

## Changelog

| Component | Changelog(CN) | Changelog(EN) | Related issues |
| --------- | ------------- | ------------- | -------------- |
|     Select      |   `Select` 修复 `allowCreate` 为 `false` 时，自动分词引入了新选项的问题。    |    `Select` fixes an issue where automatic word segmentation introduced new options when `allowCreate` was `false`.   |    Close: #463   |

## Checklist:

- [x] Test suite passes (`npm run test`)
- [ ] Provide changelog for relevant changes (e.g. bug fixes and new features) if applicable.
- [ ] Changes are submitted to the appropriate branch (e.g. features should be submitted to `feature` branch and others should be submitted to `main` branch)

### CODE REVIEW

File: components/Select/select.tsx [hunk 1]
```diff
@@ -552,7 +552,7 @@ function Select(baseProps: SelectProps, ref) {
 
   const handleTokenSeparators = (str): boolean => {
     let hasSeparator = false;
-    if (isMultipleMode && isArray(tokenSeparators) && tokenSeparators.length) {
+    if (isMultipleMode && isArray(tokenSeparators) && tokenSeparators.length && allowCreate) {
```
[the remark this answers is not in the document]
Comment (chenc041) [reply components/Select/select.tsx:555]: ok ~

File: components/Select/select.tsx [hunk 2]
```diff
@@ -565,7 +565,7 @@ function Select(baseProps: SelectProps, ref) {
           if (!optionInfoMap.get(v) || v === inputValue) {
```
Comment (MisterLuffy) [inline components/Select/select.tsx:565]: 这个 if 分支的代码可以删除了，否则在 `allowCreate = false` 时可能会多触发一次 `onChange` 回调。

Comment (MisterLuffy) [reply components/Select/select.tsx:565]: 我帮你删吧~ 准备发版了

Comment (chenc041) [reply components/Select/select.tsx:565]: 哈哈哈 好的
怪不的 今天关了这么多 pr 原来是要发版了

**Pull request merged**